### PR TITLE
64864 Update the Start Scheduling button to an action link

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
This PR updates the 'Start scheduling' button to an action link component.

## Related issue(s)

department-of-veterans-affairs/va.gov-team#64864

## Testing done
Unit and e2e testing.

## Screenshots
localhost_3001_my-health_appointments_

## GA Analytics Events
[
    {
        "event": "login-account-transition-success-modal-closed"
    },
    {
        "event": "phased-roll-out-enabled",
        "product-description": "Header V2"
    },
    {
        "event": "phased-roll-out-enabled",
        "product-description": "VAOS - Facility Selection v2"
    },
    {
        "event": "api_call",
        "api-name": "GET /v0/user",
        "api-status": "successful"
    },
    {
        "event": "vaos-new-appointment-list"
    },
    {
        "event": "vaos-get-future-appointments-started"
    },
    {
        "event": "vaos-get-pending-appointments-started"
    },
    {
        "event": "vaos-get-pending-appointments-retrieved"
    },
    {
        "event": "vaos-get-future-appointments-retrieved"
    },
    {
        "event": "vaos-number-of-items-retrieved",
        "vaos-item-type": "upcoming",
        "vaos-number-of-items": 24
    },
    {},
    {
        "event": "vaos-number-of-items-retrieved",
        "vaos-item-type": "video_home",
        "vaos-number-of-items": 2
    },
    {},
    {
        "event": "vaos-number-of-items-retrieved",
        "vaos-item-type": "video_atlas",
        "vaos-number-of-items": 1
    },
    {},
    {
        "event": "vaos-number-of-items-retrieved",
        "vaos-item-type": "video_va_facility",
        "vaos-number-of-items": 0
    },
    {},
    {
        "event": "vaos-number-of-items-retrieved",
        "vaos-item-type": "video_gfe",
        "vaos-number-of-items": 0
    },
    {},
    {
        "event": "vaos-number-of-items-retrieved",
        "vaos-item-type": "video_store_forward",
        "vaos-number-of-items": 0
    },
    {},
    {
        **"event": "vaos-vaos-start-scheduling-link"**
    }
]
## What areas of the site does it impact?
VAOS

## Acceptance criteria

- [ ] Given the user is authenticated to the Appointments tool
When the user lands on the Appointments landing page
Then the user will see the updated link to Start scheduling workflow
And page design matches the design spec

- [ ] Given the user is authenticated to the Appointments tool
When the user lands on the Appointment landing page and clicks on the Start scheduling link
Then an event will trigger vaos-start-scheduling-link in UA (Universal Analytics)

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx [hunk 1]
```diff
@@ -27,8 +31,20 @@ function ScheduleNewAppointmentButton() {
...
   const { typeOfCare } = useSelector(getNewAppointmentFlow);
+  const featureStartSchedulingLink = useSelector(
+    selectFeatureStartSchedulingLink,
+  );
 
-  return (
+  return featureStartSchedulingLink ? (
+    // eslint-disable-next-line jsx-a11y/anchor-is-valid
+    <a
+      className="vads-c-action-link--green vads-u-margin-bottom--2p5"
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx:41]: May want to add the css class name `vaos-hide-for-print` to hide the action link in the print preview.

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx:41]: Thanks Jenny!!!

File: src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx [hunk 2]
```diff
@@ -8,12 +8,16 @@ import {
...
 
 function handleClick(history, dispatch, typeOfCare) {
-  return () => {
+  return e => {
+    // Stop default behavior for anchor tag since we are using React routing.
+    e.preventDefault();
+
     recordEvent({
       event: `${GA_PREFIX}-schedule-appointment-button-clicked`,
     });
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx:23]: Based on the AC, if the `va_online_scheduling_start_scheduling_link` flag is on, it should record the event `vaos-start-scheduling-link` in UA (Universal Analytics)

**Review by jenniemc — COMMENTED** [review]
One more change.

File: src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx:31]: Sorry, moving to fast!

File: src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx [hunk 3]
```diff
@@ -8,14 +8,26 @@ import {
...
+  featureBreadcrumbUrlUpdate = false,
+) {
+  return e => {
+    // Stop default behavior for anchor tag since we are using React routing.
+    e.preventDefault();
+
     recordEvent({
-      event: `${GA_PREFIX}-schedule-appointment-button-clicked`,
+      event: featureBreadcrumbUrlUpdate
+        ? `${GA_PREFIX}-vaos-start-scheduling-link`
```
Comment (jenniemc) [inline src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx:29]: It is recording the event as `vaos-vaos-start-scheduling-link`. Suggest removing "-vaos"
```suggestion
        ? `${GA_PREFIX}-start-scheduling-link`
```

**Review by jenniemc — APPROVED** [review]
LGTM!

**Pull request merged**